Sprint 9 solution http api

sibMT opened the pull request.

sibMT committed:
Реализация тз, спринт 6 с учетом стиля(17 вариант)

sibMT committed:
Реализация тз, спринт 6 с учетом стиля(18 вариант) с доработкой замечания

sibMT committed:
Реализация тз, спринт 6 с учетом стиля(18 вариант) с доработкой замечания

sibMT committed:
Спринт 7

sibMT committed:
Спринт 7(с codeStyle)

sibMT committed:
Спринт 7(с codeStyle)

sibMT committed:
Спринт 7(с доработкой замечаний)

sibMT committed:
Спринт 8

sibMT committed:
Спринт 8

sibMT committed:
Спринт 8

sibMT committed:
Спринт 8

sibMT committed:
Спринт 8(c доработками)

sibMT committed:
Спринт 9

sibMT committed:
Спринт 9

sibMT committed:
Спринт 9

sibMT committed:
Спринт 9

sibMT committed:
Спринт 9

sibMT committed:
Изнуряющий Спринт 9

sibMT committed:
Изнуряющий Спринт 9

sibMT committed:
Изнуряющий Спринт 9

avfyodorov commented on src/server/HttpTaskServer.java:
```diff
@@ -0,0 +1,67 @@
+        InetSocketAddress address = new InetSocketAddress(HOST, PORT);
+        httpServer = HttpServer.create(address, 0);
+
+        Gson gson = new GsonBuilder()
```
gson можно было бы создавать прямо в базовом обработчике, тогда не пришлось бы его тащить вторым параметром в классы-наследники. 
На усмотрение

sibMT replied:
Точно, что-то не догадался до такого, спасибо

avfyodorov commented on src/server/handler/TaskHandler.java:
```diff
@@ -0,0 +1,209 @@
+            return false;
+        }
+
+        if (task.getStartTime() != null) {
```
Тут нужна ещё проверка на пересечение по времени.  По Тз:
.---------
если добавляемая задача пересекается с существующими — статус 406 (Not Acceptable);

avfyodorov commented on src/server/handler/SubtaskHandler.java:
```diff
@@ -0,0 +1,221 @@
+        return gson.fromJson(json, Subtask.class);
+    }
+
+    private boolean isValidSubtask(Subtask subtask) {
```
И здесь нужна проверка на пересечение по времени.  По Тз:
.---------
если добавляемая задача пересекается с существующими — статус 406 (Not Acceptable);

avfyodorov reviewed (commented):
Добрый день, Максим!

Практически всё сделано, осталось чуть  поправить.

sibMT committed:
Изнуряющий Спринт 9
Не стал менять параметр конструктора ValidationException. Написал 406 в виде String

sibMT committed:
Изнуряющий Спринт 9
Не стал менять параметр конструктора ValidationException. Написал 406 в виде String

sibMT committed:
Изнуряющий Спринт 9
Не стал менять параметр конструктора ValidationException. Написал 406 в виде String

sibMT committed:
Изнуряющий Спринт 9
Не стал менять параметр конструктора ValidationException. Написал 406 в виде String

sibMT committed:
Изнуряющий Спринт 9
Не стал менять параметр конструктора ValidationException. Написал 406 в виде String

avfyodorov reviewed (commented):
Добрый день, Максим!

Замечаний нет.
Работа принята.
